src/tests/unit/tests/assessments/assessments-requirements-filter.test.ts

```diff
@@ -0,0 +1,64 @@
+import { AssessmentsProvider } from 'assessments/types/assessments-provider';
+import { Assessment } from 'assessments/types/iassessment';
+
+describe('filter by requirements', () => {
+    const assessments: Assessment[] = [
+        {
+            key: 'x',
+            featureFlag: { required: ['x'] },
+            requirements: [{ key: 'a' }, { key: 'b' }],
+        } as Assessment,
+        { key: 'y', featureFlag: { required: ['y'] }, requirements: [{ key: 'c' }] } as Assessment,
+        {
+            key: 'x & y',
+            featureFlag: { required: ['x', 'y'] },
+            requirements: [{ key: 'd' }, { key: 'e' }],
+        } as Assessment,
+        { key: 'empty', featureFlag: { required: [] }, requirements: [] } as Assessment,
+        { key: 'missing', requirements: [{ key: 'f' }, { key: 'g' }] } as Assessment,
+    ];
+
+    let baseProvider: AssessmentsProvider;
+    let create: (flags, requirements) => AssessmentsProvider;
+
+    beforeEach(() => {
+        baseProvider = AssessmentsProviderImpl.Create(assessments);
+        create = (flags, requirements) =>
+            assessmentsProviderForRequirements(baseProvider, flags, requirements);
+    });
+
+    it.each`
+        name                                         | requirements                      | flags
+        ${'only one requirement'}                    | ${['c']}                          | ${{ x: true, y: true }}
+        ${'all requirements'}                        | ${['a', 'b', 'c', 'd', 'e', 'f']} | ${{ x: true, y: true }}
+        ${'all requirements with featureFlag false'} | ${['a', 'b', 'c', 'd', 'e', 'f']} | ${{ x: false, y: true }}
+        ${'all requirements with no feature flags'}  | ${['a', 'b', 'c', 'd', 'e', 'f']} | ${{}}
+        ${'empty requirements'}                      | ${[]}                             | ${{ x: true, y: true }}
+    `(
+        'for case $name contains only automated checks and assessments with the included requirements',
+        ({ flags, requirements }) => {
+            const testSubject = create(flags, requirements);
+            const containsOnlyIncludedRequirements = assessment => {
+                return assessment.key === 'automated-checks'
+                    ? true
+                    : assessment.requirements.every(requirement =>
+                          requirements.includes(requirement.key),
+                      );
+            };
+            const excludesDisabledFlags = assessment => {
+                return assessment.featureFlag
+                    ? assessment.featureFlag.required.every(key => flags[key] === true)
+                    : true;
+            };
+            expect(testSubject.forKey('automated-checks')).toStrictEqual(AutomatedChecks);
+            expect(testSubject.all().every(containsOnlyIncludedRequirements)).toBe(true);
+            expect(testSubject.all().every(excludesDisabledFlags)).toBe(true);
+            expect(testSubject.all()).toMatchSnapshot();
```

Comment:
I wonder if it makes sense to only check for existence of key here. The snapshot of this is absolutely massive and we don't really need to validate that all the data made it for this to be a valuable test. I believe this is also what the feature flag filter class does.